kpops/pipeline_generator/pipeline.py

```diff
@@ -229,45 +273,64 @@ def __str__(self) -> str:
             )
         )
 
-    @staticmethod
-    def substitute_component_specific_variables(
-        component: PipelineComponent, pair: dict  # TODO: better parameter name for pair
-    ) -> dict:
-        """Overrides component config with component config in pipeline environment definition."""
-        # HACK: why do we need an intermediate JSON object?
-        component_data: dict = json.loads(
-            substitute(
-                json.dumps(pair),
-                {
-                    "component_type": component.type,
-                    "component_name": component.name,
-                },
+    def substitute_in_component(self, component_as_dict: dict) -> dict:
+        """Substitute all $-placeholders in a component in dict representation
+
+        :param component_as_dict: Component represented as dict
+        :type component_as_dict: dict
+        :return: Updated component
+        :rtype: dict
+        """
+        config = self.config
+        # Leftover variables that were previously introduced in the component by the substitution
+        # functions, still hardcoded, because of their names.
+        # TODO: Get rid of them with other breaking changes
```

Comment:
would it be possible to change the config option from default_error_topic_name → error_topic_name and avoid a breaking change by setting alias="default_error_topic_name"?

Reply:
ok, then nvm for now

Reply:
I actually have a proposed work around - the user could alias any variables as they desire. Here is an example:

`export output_topic_name = topic_name_config_default_output_topic_name`
`export error_topic_name = topic_name_config_default_error_topic_name`

Reply:
#234